src/postamble_minimal.js

```diff
@@ -106,7 +106,13 @@ var imports = {
   'a': wasmImports,
 #else // MINIFY_WASM_IMPORTED_MODULES
   'env': wasmImports,
+
+// There does not currently exist a ENVIRONMENT_MAY_BE_WASI, so liken it to the shell environments.
+// (e.g. anyone building with -sENVIRONMENT=web won't be running in WASI)
+#if ENVIRONMENT_MAY_BE_NODE || ENVIRONMENT_MAY_BE_SHELL
   '{{{ WASI_MODULE_NAME }}}': wasmImports,
```

Comment:
I believe there are parts of the C library that depend on WASI-named syscalls.

We could perhaps see if any of the imports come from the WASI namespace and make this line conditional (for both minimal and regular runtime).

Reply:
Also, won't those that care about code size already fall into `MINIFY_WASM_IMPORTED_MODULES` above, avoiding this code block completely?

Reply:
> Also, won't those that care about code size already fall into `MINIFY_WASM_IMPORTED_MODULES` above, avoiding this code block completely?

This only happens if METADCE is being run, which it isn't if ASYNCIFY is enabled.